lib/mix/test/mix/tasks/help_test.exs

```diff
@@ -43,6 +57,64 @@ defmodule Mix.Tasks.HelpTest do
     end)
   end
 
+  defmodule ComplexAliases do
+    def project do
+      [
+        aliases: [
+          h: "hello",
+          p: &inspect/1,
+          help: ["help", "hello"],
+          "nested.h": [&Mix.shell().info(inspect(&1)), "h foo bar"]
+        ]
+      ]
+    end
+  end
+
+  test "help ALIAS", context do
+    Mix.Project.push(ComplexAliases)
+
+    in_tmp(context.test, fn ->
+      output =
+        capture_io(fn ->
+          Mix.Tasks.Help.run(["h"])
+        end)
+
+      assert output =~ "# mix h\n\n"
+      assert output =~ "Alias for \"hello\"\n"
+      assert output =~ ~r/^Location: mix.exs/m
+
+      output =
+        capture_io(fn ->
+          Mix.Tasks.Help.run(["p"])
+        end)
+
+      assert output =~ "# mix p\n\n"
+      assert output =~ "Alias for &Kernel.inspect/1\n"
+      assert output =~ ~r/^Location: mix.exs/m
+
+      output =
+        capture_io(fn ->
+          Mix.Tasks.Help.run(["help"])
+        end)
+
+      assert output =~ "# mix help\n\n"
+      assert output =~ "Alias for [\"help\", \"hello\"]\n"
+      assert output =~ ~r/^Location: mix.exs/m
+
+      output =
+        capture_io(fn ->
+          Mix.Tasks.Help.run(["nested.h"])
+        end)
+
+      assert output =~ "# mix nested.h\n\n"
+
+      assert output =~
+               ~r/Alias for \[#Function<.*\/1 in .*ComplexAliases.project\/0>, \"h foo bar\"\]\n/
```

Comment:
Couldn't find an easy way to format the case for `"nested.h": [&Mix.shell().info(inspect(&1)), "h foo bar"]`.
Thoughts?

Reply:
@Sihui this is honestly good enough to me. :)